awx/main/utils/ansible.py

```diff
@@ -97,3 +97,72 @@ def could_be_inventory(project_path, dir_path, filename):
     except IOError:
         return None
     return inventory_rel_path
+
+
+# This method is copied directly from Ansible core code base
+# lib/ansible/module_utils/json_utils.py
+# For purpose, see: https://github.com/ansible/ansible/issues/50100
+# Any patches to this method should sync from that version
+# NB: a copy of this function exists in ../../modules/core/async_wrapper.py. Ensure any
+# changes are propagated there.
+def _filter_non_json_lines(data):
```

Comment:
No, I really don't want to do this :(

Reply:
Isn't this function - based on my initial read - going to make certain inventory sources with _lots_ of verbose output (Satellite6 comes to mind which frequently outputs 100s of MB in verbose modes) going to make stdout ingestion _miserably_ slow?

Reply:
At the _very_ least, can we limit this behavior to plugins that won't guarantee JSON output on stdout (OpenStack)?

Reply:
limiting the behavior was my intent with the wrapping function. Two simple checks of `endswith` and `startswith` shouldn't be duplicating the string, I don't know. But yes, that's a very very good point. I am kind of concerned about the case of _middle of the road_ verbosity with large outputs. The simple fact that verbosity is on dictates a necessity to do the expensive splitting, so that may need some guard rails. It's true that this is mainly useful on failed imports, and opposed to those that will be successful.... but success/failed state is hard to predict in advance.

Reply:
At the end of the day, _in my opinion_, this is a failing of `ansible-inventory` as a tool.  If we can't rely on `ansible-inventory --list` (or perhaps some other CLI call) to **always, reliably** print valid JSON, we're going to keep dealing with this everywhere.